### PR TITLE
Woo: Product Create to use data layer requests

### PR DESCRIPTION
**Pull request by coderkevin**
This alters the Product Create code to use the data layer requests for a
more declarative approach to requesting API calls.

To Test: Run `npm test` and ensure all tests pass.

### CODE REVIEW

File: client/extensions/woocommerce/state/sites/products/actions.js [hunk 1]
```diff
@@ -12,10 +12,10 @@ import {
  * @param {Number} siteId The id of the site upon which to create the product.
  * @param {Object} product The complete product object (may include a placeholder id)
  * @param {String} [successAction=undefined] Optional action object to be dispatched upon success.
- * @param {String} [errorAction=undefined] Optional action object to be dispatched upon error.
+ * @param {String} [failureAction=undefined] Optional action object to be dispatched upon error.
  * @return {Object} Action object
  */
-export function createProduct( siteId, product, successAction = undefined, errorAction = undefined ) {
```
Comment (DanReyLop) [inline client/extensions/woocommerce/state/sites/products/actions.js]: There's no need for the `=undefined`, it's implicit.

Also, there's no need for the `if ( failureAction ) { action.failureAction = failureAction }`. Just set `action.failureAction` always. If `failureAction` is `undefined`, the resulting object will be the same (for our purposes).

**Comment by DanReyLop** [comment]
You have lint errors, please don't forget to take a look at them :)

**Comment by coderkevin** [comment]
Rebased and fixed a lint error.

File: client/extensions/woocommerce/state/data-layer/products/index.js [hunk 2]
```diff
@@ -23,58 +21,10 @@ function createProduct( { dispatch }, action, next ) {
...
-	}
-}
-
-function isValidProduct( product ) {
-	return (
-		product &&
-		product.id && ( 'number' === typeof product.id ) &&
-		product.type && ( 'string' === typeof product.type )
-	);
+	dispatch( post( siteId, 'products', productData, successAction, failureAction ) );
```
Comment (justinshreve) [inline client/extensions/woocommerce/state/data-layer/products/index.js]: 🎉 yay for red changesets -- especially when they become 1 line.

**Pull request merged**